plugins/extend-give-wp/src/admin/views/option-page-template.php

```diff
@@ -5,7 +5,7 @@
 ?>
 <div class="wrap">
     <h1><?php echo esc_html( get_admin_page_title() ); ?></h1>
-    <form action="<?php menu_page_url( 'extend-give-wp-options' ); ?>" method="post">
+    <form action="<?php esc_url( menu_page_url( 'extend-give-wp-options' ) ); ?>" method="post">
```

Comment:
@rgadon107 You need to `echo` the escaped URL for it to appear in the view.

Reply:
@hellofromtonya 
>You need to echo the escaped URL for it to appear in the view.

I don't think that's true. When I add `echo esc_url()`, the url path gets rendered twice in the web inspector. See comparison below.

>Screenshot 1 ( esc_url is NOT echoed)
>HTML: `<form action="<?php esc_url( menu_page_url( 'extend-give-wp-options' ) ); ?>" method="post">`

![esc_url is not echoed out](https://user-images.githubusercontent.com/3718603/72293716-b9f2c980-3619-11ea-972d-816ac5733f76.png)

>Screenshot 2 ( esc_url IS echoed ). 
>HTML: `<form action="<?php echo esc_url( menu_page_url( 'extend-give-wp-options' ) ); ?>" method="post">`

![esc_url IS echoed out](https://user-images.githubusercontent.com/3718603/72293882-08a06380-361a-11ea-8da9-77678d0cbc28.png)

Reply:
The reason is it's rendering straight out of `menu_page_url()` instead of passing through `esc_url()`. You want to pass in `false` as the 2nd argument for `menu_page_url()`. [Reference here.](https://developer.wordpress.org/reference/functions/menu_page_url/)

Reply:
@hellofromtonya 

>`echo` param = false.
>`<form action="<?php esc_url( menu_page_url( 'extend-give-wp-options', false ) ); ?>" method="post">`
>Web inspector of option input form. 'action` attribute is empty. 
![menu_page_url with echo equals false](https://user-images.githubusercontent.com/3718603/72296537-7602c300-361f-11ea-8d15-f882bca2c573.png)

>`echo` param = true (default)
>`<form action="<?php esc_url( menu_page_url( 'extend-give-wp-options', true
>) ); ?>" method="post">`
>Web inspector of option input form. 'action` attribute contains value. 
![menu_page_url with echo equals true](https://user-images.githubusercontent.com/3718603/72296695-ddb90e00-361f-11ea-8ace-652d250dc5a7.png)

Don't we want `$echo` set to `true`?

Reply:
>Don't we want $echo set to true?

No. Why? If echo is `true`, then it renders it instead of returning it back to be further processed by `esc_url()`. In other words, `esc_url()` doesn't run as nothing is passed into it.

Reply:
Note: I misunderstood what Tonya was asking for here. It should have been: 

>`<form action="<?php echo esc_url( menu_page_url( 'extend-give-wp-options', false ) ); ?>" method="post">`

As it turn out, the markup was simply replaced with the string 'options.php'. The final markup was: 

>`<form action="options.php" method="post">`

Since the admin menu form renders on 'options-general.php', WordPress redirected the option template form to the correct page.